### PR TITLE
feat: update events docs

### PR DESCRIPTION
**Pull request by jyecusch**

### CODE REVIEW

File: docs/reference/api/events/subscribe.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+Resources that can subscribe to a topic, such as [functions](/docs/intro/concepts/functions) and containers provide a `triggers` property defining their triggers. Adding a new `topics` object under `triggers`, followed by the name of the topic(s) to subscribe to will set up a subscription between the resource and the topic(s). In the case of functions, this causes the function to be invoked each time a new event is published to the topic. The function will receive a trigger with the event being provided as its payload. This is a useful mechanism when one service needs to communicate with another in order to have the second service perform some work.
+
+```yaml
+topics:
+	# A topic for events about newly registered users
+	user-registered: {}
+
+services:
```
Comment (tjholm) [inline docs/reference/api/events/subscribe.mdx:12]: functions

**Review by tjholm — CHANGES_REQUESTED** [review]

File: docs/reference/api/events/subscribe.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  # Registers new users and publishes an event
+  create-user:
+    handler: functions/user/register.ts
+  # Subscribes to all new user registrations and sends them a welcome email
+  welcome-user:
+    handler: functions/user/welcome.ts
+    triggers:
+      topics:
+        - user-registered
```
Comment (tjholm) [inline docs/reference/api/events/subscribe.mdx:21]: I think this is a good example of duplicate content (a positive example that is), where the api section provides a call to action, but this same content will also appear under the function/container references in the project configuration above (as documented properties of those resources). But in this case we can just provide a link from the trigger property as well.

File: docs/reference/api/events/publish.mdx [hunk 3]
```diff
@@ -74,4 +74,6 @@ It's essential that every event can be uniquely identified so that subscribers c
...
+When implementing design patterns that allow for atomic state changes and event publishing, such as [Transactional Outbox](https://microservices.io/patterns/data/transactional-outbox.html), it's critical that any attempt to 'republish' an event due to a failure reuses the same event ID as any previous attempts. This ensures duplicate event deliveries won't contain mismatching IDs in the case that a previous attempt silently succeeded, while appearing to fail. In these cases we recommend generating your own event ID and storing it in the outbox record. Ensuring depulicate events will contain matching keys is essential to allow for [idempotent subscribers]().
```
Comment (HomelessDinosaur) [inline docs/reference/api/events/publish.mdx:77]: depulicate -> duplicate

**Review by tjholm — APPROVED** [review]

File: docs/reference/api/events/publish.mdx (hunk 3 above)
Comment (jyecusch) [reply docs/reference/api/events/publish.mdx:77]: fixed

File: docs/reference/api/events/_index.mdx [hunk 4]
```diff
@@ -15,7 +15,17 @@ The Nitric events API provides an interface to message-oriented middleware and m
 ## Key concepts
 
 - **Topics** are a named resource and target for publishing events. They typically define the _subject_ of the messages being published and subscribed.
-- **Events** are a combination of metadata and a payload, produced by publishers and delivered to subscribers. Typically, these take the form of a JSON document with a standard format.
+- **Events** (_Messages_) are a combination of metadata and a payload, produced by publishers and delivered to subscribers. Typically, these take the form of a JSON document with a standard format.
 - **Publishers** produce events by sending them to a topic. Publishers are unaware of the topic's subscribers, providing strong decoupling.
 - **Subscriptions** define the subscribers of a particular topic. Each subscription defines a subscriber, and the topic they subscribe to.
 - **Subscribers** are consumers of events, typically in the form of a request that must handled.
```
Comment (medgar-nitric) [inline docs/reference/api/events/_index.mdx:21]: must **be** handled.

**Review by medgar-nitric — CHANGES_REQUESTED** [review]
Looks good, just minor typo.

File: docs/reference/api/events/_index.mdx (hunk 4 above)
Comment (jyecusch) [reply docs/reference/api/events/_index.mdx]: fixed

File: docs/reference/api/events/subscribe.mdx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+functions:
+  # Registers new users and publishes an event
+  create-user:
+    handler: functions/user/register.ts
+  # Subscribes to all new user registrations and sends them a welcome email
+  welcome-user:
+    handler: functions/user/welcome.ts
+    triggers:
+      topics:
+        - user-registered
```
[the remark this answers is not in the document]
Comment (jyecusch) [reply docs/reference/api/events/subscribe.mdx]: fixed

**Review by davemooreuws — APPROVED** [review]

**Review by tjholm — APPROVED** [review]

File: docs/reference/api/events/_index.mdx [hunk 6]
```diff
@@ -2,9 +2,9 @@
 title: 'Events Overview'
 ---
 
-Nitric's events service and API provides an intuitive approach to asynchronous messaging allowing services, functions and more to communicate via topics with event messages. While synchronous request/response communication is quite common, events provide a more reliable mechanism for communication in distributed and serverless systems, with the added benefit that services can be decoupled from each other through a [publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) relationship.
+Events are a form of asynchronous messaging which enable functions and other stack resources to communicate with one another. Nitric Events follow a [publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) model, where messages are published to Topics by Publishers and forwarded to Subscribers. While synchronous request/response communication is quite common, events assist in decoupling services from each other and provide mechanisms that improve fault tolerance. These benefits are important in the creation of distributed and serverless systems, where functions and services are often stateless and ephemral.
```
Comment (medgar-nitric) [inline docs/reference/api/events/_index.mdx:6]: There are other benefits of pub/sub architectures, temporal decoupling, load leveling, efficient resource use, etc., but this is probably not the righ place to go into all this.

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**